src/spyglass/common/common_optogenetics.py

```diff
@@ -0,0 +1,377 @@
+            virus injection data.
+        """
+        key = dict(
+            nwb_file_name=nwb_file_name,
+            virus_name=virus_injection_object.virus.construct_name,
+            injection_object_id=virus_injection_object.object_id,
+            name=virus_injection_object.name,
+            description=virus_injection_object.description,
+            hemisphere=virus_injection_object.hemisphere,
+            location=virus_injection_object.location,
+            ap_location=virus_injection_object.ap_in_mm,
+            ml_location=virus_injection_object.ml_in_mm,
+            dv_location=virus_injection_object.dv_in_mm,
+            pitch=virus_injection_object.pitch_in_deg,
+            roll=virus_injection_object.roll_in_deg,
+            yaw=virus_injection_object.yaw_in_deg,
+            volume=virus_injection_object.volume_in_uL,
+            titer=virus_injection_object.virus.titer_in_vg_per_ml,
+        )
+
+        Virus().insert_from_nwb_object(virus_injection_object.virus)
+        self.insert1(
+            key,
+        )
+
+    def make(self, key):
+        # for use with populate_all_common
+        nwb = (Nwbfile() & key).fetch_nwb()[0]
+        for obj in nwb.objects.values():
+            if isinstance(obj, OptogeneticVirusInjection):
+                self.insert_from_nwb_object(key["nwb_file_name"], obj)
+
+
+@schema
+class OpticalFiberDevice(SpyglassMixin, dj.Manual):
+    definition = """
+    # Optical fiber device information
+    fiber_name: varchar(255)  # name of the device
+    ---
+    model: varchar(255)  # model of the device
+    manufacturer: varchar(255)
+    numerical_aperture: float  # numerical aperture of the fiber
+    core_diameter: float # core diameter of the fiber (in um)
+    active_length: float # active length of the fiber (in mm)
+    ferrule_name: varchar(255) # name of the ferrule
+    ferrule_diameter: float # diameter of the ferrule (in mm)
+    """
+
+    def insert_from_nwb_object(self, fiber_object):
+        """
+        Insert an OpticalFiberDevice entry from an NWB object.
+
+        Parameters
+        ----------
+        fiber_object : OpticalFiberDevice
+            An instance of the OpticalFiberDevice class containing the fiber
+            device data.
+        """
+        key = dict(
+            fiber_name=fiber_object.model.fiber_name,
+            model=fiber_object.model.fiber_model,
+            manufacturer=fiber_object.model.manufacturer,
+            numerical_aperture=fiber_object.model.numerical_aperture,
+            core_diameter=fiber_object.model.core_diameter_in_um,
+            active_length=fiber_object.model.active_length_in_mm,
+            # Remove non-printable ASCII characters from ferrule_name to ensure database compatibility
+            ferrule_name="".join(
+                c
+                for c in fiber_object.model.ferrule_name
+                if c in string.printable
```

Comment:
I'm not familiar with what these would entail, but do we need a nearest-neighbor to make sure it looks right on fetch?

Reply:
The question I had was more about fetching. What kind of characters get excluded? Will it be readable when retrieved?